src/phase0/state_transition/block_processing.rs

```diff
@@ -160,16 +163,38 @@ fn process_block_header<
         MAX_VALIDATORS_PER_COMMITTEE,
         PENDING_ATTESTATIONS_BOUND,
     >,
-    _block: &mut BeaconBlock<
+    block: &mut BeaconBlock<
         MAX_PROPOSER_SLASHINGS,
         MAX_VALIDATORS_PER_COMMITTEE,
         MAX_ATTESTER_SLASHINGS,
         MAX_ATTESTATIONS,
         MAX_DEPOSITS,
         MAX_VOLUNTARY_EXITS,
     >,
-    _context: &Context,
+    context: &Context,
 ) -> Result<(), Error> {
+    if block.slot != state.slot {
+        return Err(Error::InvalidHeader(InvalidHeader::StateSlotMismatch));
+    }
+
```

Comment:
missing some validations here:

- block slot is newer than the latest header in the state
- block's proposer is equal to the one expected based on our local state

Reply:
My bad, I was looking at the old version of the specification. Added here: 7b8dfbf